core/src/main/java/io/stargate/core/metrics/StargateTenantExtractor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.stargate.core.metrics;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StargateTenantExtractor {
+
+  private static final Pattern TENANT_PATTERN =
+      Pattern.compile("[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}");
+
+  private static final String[] HOST_HEADERS = new String[] {"x-forwarded-host", "x-host"};
+
+  /**
+   * Returns tenant tag from the headers. If not found returns <code>null</code>.
+   *
+   * @param headers Header map
+   * @return Tag or null
+   */
+  public static String tenantFromHeaders(Map<String, List<String>> headers) {
```

Comment:
Could this be refactored into an OSGi service? Perhaps something like `TagsExtractor` returning a `Map` of `tag_name` -> `value`?

Reply:
We could define an interface for extracting tags from HTTP headers and _no_ default impl.

In environments where this is relevant, the impl. will have to be registered in OSGi. Our metrics handling code will look it up, and use if available. 

We could use a naming convention for service identifiers similar to what we have for `Persistence` to allow different implementations to co-exist without conflicts. The "active" impl. could be selected by a java system property.

Reply:
This seems to be conceptually similar to `AuthorizationProcessor`

Reply:
Done as we agreed.